File: container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
      if(height.length==2){
          return Math.min(height[0],height[1]);
      }
        int i=0;
        int f=height.length-1;
        int max=0;
        while(i<f){
            int dist=f-i;
            int min=Math.min(height[i],height[f]);
            max=Math.max(max,dist*min);
            if(height[i]<height[f]){
                i++;
            }
            else{
                f--;
            }
        }
        return max;
    }
}

File: container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l=0
        r=len(height)-1
        max_len=0
        while l<r:
            length=min(height[l],height[r])
            dist=r-l
            max_len=max(max_len,length*dist)
            if height[l]>=height[r]:
                r-=1
            elif height[l]<height[r]:
                l+=1
            
        return max_len
